### PR TITLE
(MODULES-4294) WIP adds i18n decorator for ruby

### PR DESCRIPTION
**Pull request by eputnam**
DEPENDS ON #729 

All appropriate strings in lib/facter decorated with the _() function
All appropriate strings in lib/puppet/type and lib/puppet/provider wrapped in _() function
All appropriate strings in lib/puppet/parser/functions wrapped with _() function
All appropriate strings in lib/puppet/functions wrapped with _() function

### CODE REVIEW

**Comment by tphoney** [comment]
looks good,  how did you change so much ? through an ide or sed/awk ?

**Comment by pmcmaw** [comment]
@tphoney The changes were carried out using the following script:
https://github.com/eputnam/string-wrapper

File: lib/facter/facter_dot_d.rb [hunk 1]
```diff
@@ -49,7 +49,7 @@ def txt_parser(file)
       end
     end
   rescue StandardError => e
-    Facter.warn("Failed to handle #{file} as text facts: #{e.class}: #{e}")
+    Facter.warn(_("Failed to handle #{file} as text facts: #{e.class}: #{e}"))
```
Comment (DavidS) [inline lib/facter/facter_dot_d.rb]: This will not create a translatable string, as it interpolates `file` and `e.class` before looking up the translation.

Comment (eputnam) [inline lib/facter/facter_dot_d.rb]: is this true for all the Facter messaging?

Comment (DavidS) [inline lib/facter/facter_dot_d.rb]: This is true for all strings.

Comment (DavidS) [inline lib/facter/facter_dot_d.rb]: See https://github.com/puppetlabs/gettext-setup-gem#interpolation for background.

Comment (eputnam) [inline lib/facter/facter_dot_d.rb]: i've been told interpolated strings are a non-issue which is why i thought i could get away with it. i'll be looking into this today.

**Comment by glennsarti** [comment]
PR #729 has been merged. Needs a rebase

**Comment by tphoney** [comment]
@eputnam is this still relevant ? can it be closed

**Comment by eputnam** [comment]
@TP this is just all strings decorated using the scripts we have currently, but if it's going to get us caught doing something we don't want to do then i guess it's not worth keeping...

**Comment by eputnam** [comment]
failures are expected on Puppet 4.2.3 which does not know how to handle the _() function

**Pull request closed without merging**